src/components/ProfileModal/ProfileModal.tsx

```diff
@@ -0,0 +1,128 @@
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+
+import { useState, useEffect } from "react";
+import { z } from "zod";
+import { useSession } from "next-auth/react";
+import { MoveLeft } from "lucide-react";
+import { changePWSchema, userSchema } from "@/utils/types";
+import ViewProfileModal from "./ViewProfileModal";
+import EditProfileModal from "./EditProfileModal";
+import ChangePasswordModal from "./ChangePasswordModal";
+
+export function ProfileModal() {
+  const [profileState, setProfileState] = useState<
+    "view" | "changePw" | "edit"
+  >("view");
+  const [open, setOpen] = useState(false);
+
+  const userDataSchema = userSchema.extend({
+    _id: z.string().length(24),
+  });
+
+  const { data: session } = useSession();
+  const currentUser = session?.user;
+  const [userData, setUserData] = useState<z.infer<typeof userDataSchema>>();
+
+  useEffect(() => {
+    if (currentUser) {
+      getUserData();
+    }
+  }, [currentUser]);
+
+  async function getUserData() {
+    try {
+      const response = await fetch(`/api/users/${currentUser?._id}`);
+      const data = await response.json();
+      setUserData(data.data);
+    } catch (error) {
+      console.error("Error getting user:", error);
+    }
+  }
+
+  async function editUser(
+    data: z.infer<typeof userDataSchema> | z.infer<typeof changePWSchema>,
+    type: "info" | "password"
+  ) {
+    try {
+      const params = new URLSearchParams({ type: type });
+
+      const response = await fetch(`/api/users/${currentUser?._id}/?${params.toString()}`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(data),
+      });
+      const responseData = await response.json();
+      return responseData;
+    } catch (error) {
+      console.error(`Error editing user ${type}:`, error);
+    }
+  }
```

Comment:
Define this outside of the function (since it doesn't use or update state), this reduces the amount of information that needs to be reprocessed on every render. This will also allow you to export the parameters and return type of this function so it can be reused across your components.